cli/src/main.rs

```diff
@@ -234,6 +229,57 @@ pub(crate) struct CoreOptions {
     core: usize,
 }
 
+#[derive(clap::Parser)]
+pub(crate) struct FormatOptions {
+    #[clap(value_enum, ignore_case = true, default_value = "elf", long)]
+    format: Format,
+    /// The address in memory where the binary will be put at.
+    #[clap(long)]
+    pub base_address: Option<u64>,
+    /// The number of bytes to skip at the start of the binary file.
+    #[clap(long, default_value = "0")]
+    pub skip: u32,
+    /// The idf bootloader path
+    #[clap(long)]
+    pub bootloader: Option<PathBuf>,
+    /// The idf partition table path
+    #[clap(long)]
+    pub partition_table: Option<PathBuf>,
+}
+
+impl FormatOptions {
+    pub fn into_format(self) -> anyhow::Result<Format> {
+        Ok(match self.format {
+            Format::Bin(_) => Format::Bin(BinOptions {
+                base_address: self.base_address,
+                skip: self.skip,
+            }),
+            Format::Hex => Format::Hex,
+            Format::Elf => Format::Elf,
+            Format::Idf(_) => {
+                let bootloader = if let Some(path) = self.bootloader {
+                    Some(std::fs::read(path)?)
+                } else {
+                    None
+                };
+
+                let partition_table = if let Some(path) = self.partition_table {
+                    Some(esp_idf_part::PartitionTable::try_from(std::fs::read(
```

Comment:
Same here as for the bootloader, I think it would fit better in `load_idf_data`.